portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,12 +40,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("text/html;");
     String greetingJson = new Gson().toJson(greetings); //set to Json
     response.getWriter().println(greetingJson); //send response
-    
-    Entity taskEntity = new Entity("Task");
-    taskEntity.setProperty("greeting", greetingJson);
-    
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); //instantiate datastoreclass
-    datastore.put(taskEntity); //put greetings in datastore 
 
   }
+
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String greetingJson = new Gson().toJson(greetings); //set to Json
```

Comment:
This will try to store the 3 greetings listed in lines 33-35, which would indeed duplicate those 3 lines.  I'm not sure how you're invoking doPost though without the HTML form to submit the info, so I'm thinking maybe I don't have the complete picture here?

If this was the issue though, I'd suggest looking through the next couple steps in the walkthrough.  Later on you'll be replacing the `greetings` part with saving the info that's contained in `request`.  This way you wouldn't be resaving what's contained in lines 33-35, but instead will be saving the info that the user entered in the form